packages/gatsby-transformer-remark/README.md

```diff
@@ -68,7 +68,29 @@ Using the following GraphQL query you'll be able to get the table of contents
     edges {
       node {
         html
-        tableOfContents(pathToSlugField: "frontmatter.path")
+        tableOfContents
+      }
+    }
+  }
+}
+```
+
+### Configuring the tableOfContents
+
+By default the tableOfContents is using the field `slug` to generate URLs. You can however provide another field using the pathToSlugField parameter. **Note** that providing a non existing field will cause the result to be null. To alter the default values for tableOfContents generation, include values for heading (string), maxDepth (number 1 to 6) and/or tight (boolean) in graphQL query.
+
+```graphql
+{
+  allMarkdownRemark {
+    edges {
+      node {
+        html
+        tableOfContents(
+          pathToSlugField: "frontmatter.path"
+          heading: "only show toc from this heading onwards"
+          maxDepth: 2
+          tight: true
```

Comment:
After reading https://www.npmjs.com/package/mdast-util-toc I still don't really understand what `heading` and `tight` options do - can you show example of using it? Or maybe add test that would show using it?

Reply:
After testing a bit, here are my insights to these two options: "heading" searches for a string in the toc headings and renders the toc from that heading onwards. Alas, I think the implementation of this feature seems to be flawed, because the toc is rendered without the given heading. The "tight" option seems to do nothing at all. As these options should be fixed in mdast-util-toc, I would omit additional tests here.

Reply:
Latest insights: "tight" influences "loose" and "spread" attributes in generated toc JSON object. The plugin that uses these attributes for html generation is "hast-util-to-html" and it toggles space after quoted attributes. Default setting for tight is false, so a space is added after quoted attributes by default (which is the desired behaviour). I'll update the readme.

Reply:
ok, I removed tight option completely, as the desired changes in html output seem to get lost somewhere along the way. Also added test for heading option.